src/sonic_ax_impl/mibs/ietf/rfc1213.py

```diff
@@ -206,17 +208,19 @@ def update_data(self):
         Update redis (caches config)
         Pulls the table references for each interface.
         """
-        self.if_counters = \
-            {sai_id: Namespace.dbs_get_all(self.db_conn, mibs.COUNTERS_DB, mibs.counter_table(sai_id), blocking=True)
-            for sai_id in self.if_id_map}
+        for sai_id_key in self.if_id_map:
+            namespace, sai_id = mibs.split_sai_id_key(sai_id_key)
+            if_idx = get_index_from_str(self.if_id_map[sai_id_key].decode())
```

Comment:
>decode [](start = 67, length = 6)

Why previous we don't need `decode()` ?

Reply:
previously we never used the if_name from if_id_map, now we use this to get the interface index.
if_id_map previously also had name in bytes.

Reply:
If previously never used, can you `decode` at the beginning of `init_sync_d_interface_tables`? So multiple `decode` could be simplified.



---
In reply to: [454632465](https://github.com/Azure/sonic-snmpagent/pull/145#discussion_r454632465) [](ancestors = 454632465)

Reply:
Correction to what I mentioned before, it was not used in this file, but rfc4363 is using this.
mibs.get_index(self.if_id_map[port_id]) . mibs.get_index() - this takes a binary string and decodes internally.
Modifying it in the init(), will impact some other users of that library function.

Reply:
Thanks for providing an exception. I checked that function, and it is also a great opportunity to simplify this way.

---
In reply to: [454665605](https://github.com/Azure/sonic-snmpagent/pull/145#discussion_r454665605) [](ancestors = 454665605,454632465)

Reply:
If we update in init_sync_d_interface_tables(), we will end up doing a decode during reinit_data(); here decode is done during update_data(), so performance will not change.

Reply:
Updated as per suggestion.